src/utils/gauges.ts

```diff
@@ -372,3 +430,171 @@ export function areGaugesActive(chainId?: ChainId): boolean {
+}
+
+const retrieveGaugeContracts = (
+  gaugeAddresses: string[],
+  chainId: ChainId,
+): MulticallContract<LiquidityGaugeV5>[] | MulticallContract<ChildGauge>[] => {
+  if (chainId === ChainId.MAINNET) {
+    return gaugeAddresses.map((address) =>
+      createMultiCallContract<LiquidityGaugeV5>(
+        address,
+        LIQUIDITY_GAUGE_V5_ABI,
+      ),
+    )
+  }
+
+  return gaugeAddresses.map((address) =>
+    createMultiCallContract<ChildGauge>(address, CHILD_GAUGE_ABI),
+  )
+}
+
+async function getSDLRates(
+  gaugeMinterContract: Minter | null,
+  gaugeRelativeWeights: BigNumber[],
+  gaugeMulticallContracts: MulticallContract<ChildGauge>[],
+  ethCallProvider: MulticallProvider,
+) {
+  if (gaugeMinterContract) {
+    const minterSDLRate = await gaugeMinterContract.rate()
+    return gaugeRelativeWeights.map((weight) =>
+      minterSDLRate.mul(weight).div(BN_1E18),
+    )
+  }
+
+  const currentTimeStamp = Date.now() / 1000
+  return await ethCallProvider.tryAll(
+    gaugeMulticallContracts.map((contract) =>
+      contract.inflation_rate(Math.floor(currentTimeStamp / 604800)),
+    ),
+  )
+}
+
+async function getGaugeRewardCounts(
+  gaugeMulticallContracts:
+    | MulticallContract<LiquidityGaugeV5>[]
+    | MulticallContract<ChildGauge>[],
+  ethCallProvider: MulticallProvider,
+) {
+  return (
+    await ethCallProvider.tryAll(
+      gaugeMulticallContracts.map((contract) => contract.reward_count()),
+    )
+  ).map((count) => count || Zero)
+}
+
+async function getGaugeRewardTokens(
+  gaugeMulticallContracts: (
+    | MulticallContract<ChildGauge>
+    | MulticallContract<LiquidityGaugeV5>
+  )[],
+  gaugeRewardCounts: BigNumber[],
+  ethCallProvider: MulticallProvider,
+) {
+  return Promise.all(
+    gaugeRewardCounts.map((count, index) =>
+      ethCallProvider.all(
+        enumerate(count.toNumber(), 0).map((num) =>
+          gaugeMulticallContracts[index].reward_tokens(num),
+        ),
+      ),
+    ),
+  )
+}
+
+async function getGaugeRewardsFromTokens(
+  gaugeRewardCounts: BigNumber[],
+  gaugeMulticallContracts: (
+    | MulticallContract<ChildGauge>
+    | MulticallContract<LiquidityGaugeV5>
+  )[],
+  gaugeRewardsTokens: string[][],
+  ethCallProvider: MulticallProvider,
+) {
+  return Promise.all(
+    gaugeRewardCounts.map((count, index) =>
+      ethCallProvider.all(
+        enumerate(count.toNumber(), 0).map(
+          (num) =>
+            gaugeMulticallContracts[index].reward_data(
+              gaugeRewardsTokens[index][num],
+            ) as unknown as MulticallCall<
+              [arg0: string, overrides?: CallOverrides | undefined],
+              [string, string, BigNumber, BigNumber, BigNumber, BigNumber] & {
+                token: string
+                distributor: string
+                period_finish: BigNumber
+                rate: BigNumber
+                last_update: BigNumber
+                integral: BigNumber
+              }
+            >,
```

Comment:
`ChildGauge.reward_data(address)` will return a different struct

```
struct Reward:
    distributor: address
    period_finish: uint256
    rate: uint256
    last_update: uint256
    integral: uint256
```

Reply:
orr maybe typescript is smart enough to fill in 0 for the missing key?

Reply:
iirc, the value comes back as either 0 or null. Either way, it hasn't caused any compatibility issues.

Reply:
My man's trying to make csv work when parquet might be a cleaner choice. `=` vs `||`